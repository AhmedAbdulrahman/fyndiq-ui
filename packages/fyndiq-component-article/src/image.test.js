import React from 'react'
import { shallow } from 'enzyme'

import Img from './image'

jest.useFakeTimers()

describe('fyndiq-component-article Image', () => {
  beforeEach(() => {
    jest.runAllTimers()
  })

  it('should be visible by default', () => {
    expect(shallow(<Img src="test.jpg" />)).toMatchSnapshot()
  })

  it('should still be visible after changing the src', () => {
    const component = shallow(<Img src="test.png" />)
    component.setProps({ src: 'test2.png' })
    expect(component.prop('style').opacity).toBe(1)
  })

  it('should not be visible after a debounce time when changing the image src', () => {
    const component = shallow(<Img src="test.png" />)
    component.setProps({ src: 'test2.png' })
    jest.runTimersToTime(10)
    component.update()
    expect(component.prop('style').opacity).toBe(0)
  })

  it('should be visible again once the image has loaded', () => {
    const component = shallow(<Img src="test.png" />)

    global.Image = function ImageMock() {
      setTimeout(() => this.onload(), 1000)
    }

    component.setProps({ src: 'test2.png' })
    jest.runTimersToTime(1000)
    component.update()
    expect(component.prop('style').opacity).toBe(1)
  })
})
