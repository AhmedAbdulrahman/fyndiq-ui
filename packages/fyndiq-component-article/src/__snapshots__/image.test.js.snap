// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fyndiq-component-article Image should be visible by default 1`] = `
<img
  src="test.jpg"
  style={
    Object {
      "opacity": 1,
      "transitionDuration": "0.3s",
      "transitionProperty": "opacity",
    }
  }
/>
`;
